### DISCUSSION TITLE
Using build function from mcr.microsoft.com/dotnet/sdk:3.1 throw error

### DISCUSSION THREAD

**Discussion by thealbres** [dotnet/dotnet-docker#3534] [Q&A]
Hi, I am trying to build a dacpac using the dotnet container, but when I try to use the `build` function it throws a error

The error I am getting is this : 
![image](https://user-images.githubusercontent.com/80134809/156418071-9bc88e55-ca2c-403c-a990-8e4963da059b.png)
`/app/Receivables.sqlproj(61,3): error MSB4019: The imported project "/usr/share/dotnet/sdk/3.1.416/Microsoft/VisualStudio/v11.0/SSDT/Microsoft.Data.Tools.Schema.SqlTasks.targets" was not found. Confirm that the expression in the Import declaration "/usr/share/dotnet/sdk/3.1.416//Microsoft/VisualStudio/v11.0/SSDT/Microsoft.Data.Tools.Schema.SqlTasks.targets" is correct, and that the file exists on disk.`

I also tried to run from command line instead of a dockerfile, same error. I found some questions on the internet about the same error, but everyone I saw about that was during the building localy, since I am using docker I don't think the approaches also work for me.

This is my docker file
```
FROM mcr.microsoft.com/dotnet/sdk:3.1 AS dacpac_builder

COPY ./../Receivables /app
RUN cd /app && dotnet build Receivables.sqlproj

```

Can someone help me? 

Thanks

**Comment by mthalman** [comment]
Your sqlproj requires VS tools in order to build it. That's not contained in the .NET Core SDK container image. Instead, you'd want to target .NET Framework SDK. While .NET Framework SDK image does contain VS tools, it doesn't include the SSDT component. To add it, you can follow the instructions at https://docs.microsoft.com/visualstudio/install/build-tools-container?view=vs-2022 which describe how to install additional components. The specific component ID which you'd want to install is `Microsoft.VisualStudio.Component.SQL.SSDTBuildSku` which I found here: https://docs.microsoft.com/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2022&preserve-view=true#data-storage-and-processing-build-tools.

**Reply by mthalman** [reply]
Sorry, forgot to mention the tag to use for the base image: `mcr.microsoft.com/dotnet/framework/sdk:4.8`

**Reply by thealbres** [reply]
@mthalman the docker file must be something like this? 

```
# escape=`

# Use the latest Windows Server Core image with .NET Framework 4.8.
FROM mcr.microsoft.com/dotnet/framework/sdk:4.8-windowsservercore-ltsc2019 AS dacpac_builder

# Restore the default Windows shell for correct batch processing.
SHELL ["cmd", "/S", "/C"]

RUN `
    # Download the Build Tools bootstrapper.
    curl -SL --output vs_buildtools.exe https://aka.ms/vs/17/release/vs_buildtools.exe `
    `
    # Install Build Tools with the Microsoft.VisualStudio.Workload.AzureBuildTools workload, excluding workloads and components with known issues.
    && (start /w vs_buildtools.exe --quiet --wait --norestart --nocache modify `
        --installPath "%ProgramFiles(x86)%\Microsoft Visual Studio\2022\BuildTools" `
        --add Microsoft.VisualStudio.Component.SQL.SSDTBuildSku `
        || IF "%ERRORLEVEL%"=="3010" EXIT 0) `
    `
    # Cleanup
    && del /q vs_buildtools.exe

# Define the entry point for the docker container.
# This entry point starts the developer command prompt and launches the PowerShell shell.
ENTRYPOINT ["C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\Common7\\Tools\\VsDevCmd.bat", "&&", "powershell.exe", "-NoLogo", "-ExecutionPolicy", "Bypass"]

COPY ./Receivables /app

RUN cd /app && dotnet build Receivables.sqlproj

```

I used the image you mentioned with the component ID `Microsoft.VisualStudio.Component.SQL.SSDTBuildSku`

But I still get the same error 
![image](https://user-images.githubusercontent.com/80134809/156691931-7943311c-7426-4832-b78f-38633e55832d.png)

Am I missing something?

**Reply by thealbres** [reply]
I also tried with `Microsoft.VisualStudio.Workload.DataBuildTools`, same error
